core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -54,22 +59,43 @@ private Fuzziness(int fuzziness) {
     }
 
     private Fuzziness(String fuzziness) {
-        if (fuzziness == null) {
+        if (fuzziness == null || fuzziness.isEmpty()) {
             throw new IllegalArgumentException("fuzziness can't be null!");
         }
         this.fuzziness = fuzziness.toUpperCase(Locale.ROOT);
     }
 
+    private Fuzziness(String fuzziness, int lowDistance, int highDistance) {
+        this(fuzziness);
+        if (lowDistance < 0 || highDistance < 0 || lowDistance > highDistance) {
+            throw new IllegalArgumentException("fuzziness wrongly configured, must be: lowDistance > 0, highDistance" +
+                " > 0 and lowDistance <= highDistance ");
+        }
+        this.lowDistance = lowDistance;
+        this.highDistance = highDistance;
+    }
+
     /**
      * Read from a stream.
      */
     public Fuzziness(StreamInput in) throws IOException {
         fuzziness = in.readString();
+        if (in.readBoolean()) {
```

Comment:
For backward compatibility of the stream format, you need to do something like this:
```
if (in.getVersion().onOrAfter(Version.V_6_1_0) && in.readBoolean()) {
  // read distances
}
```
Otherwise we might try to read a boolean from versions that do not support it.